bin/webpack-dev-server.js

```diff
@@ -455,9 +459,15 @@ function reportReadiness(uri, options) {
 	if(options.historyApiFallback)
 		console.log(`404s will fallback to ${colorInfo(useColor, options.historyApiFallback.index || "/index.html")}`);
 	if(options.open) {
-		open(uri + options.openPage).catch(function() {
-			console.log("Unable to open browser. If you are running in a headless environment, please do not use the open flag.");
-		});
+		if(argv["open"] && argv["open"] !== "") {
+			open(uri + options.openPage, { app: options.open }).catch(function() {
```

Comment:
not a fan of two nearly identical calls to `open` and subsequent logging. should be something like this:

(this is pseudo code, don't copy and paste it)
```js
var openOptions = {};
var openMessage = "Unable to open browser";

if (typeof options.open === "string") {
  openOptions = { app: options.open };
  openMessage =+ " :  " + options.open;
}

open(uri + options.openPage, openOptions).catch(function () {
  console.log(openMessage + "If you are running in a headless environment, please do not use the open flag.");
});
```